File: LSP/Ruby/não aplicado/teste.rb
require 'test/unit'

class Teste < Test::Unit::TestCase
    def area_deve_ser_6_quando_retangulo_2x3
    	retangulo = Retangulo.new
		retangulo.altura = 2
		retangulo.largura = 3
		assert_equal 6, retangulo.area
    end
    
	def area_deve_ser_9_quando_quadrado_3x3
		quadrado = Quadrado.new
		quadrado.largura = 3
		assert_equal 9, quadrado.area
    end
	
	def area_deve_ser_20_quando_quadrado_retangulo_4x5
		quadrado = Quadrado.new
		quadrado.altura = 4
		quadrado.largura = 5
		assert_equal 20, quadrado.area
    end
end

File: LSP/Delphi/não aplicado/Teste.pas
unit Teste;

interface

uses
  TestFramework, Unit1;

type
  TestTRetangulo = class(TTestCase)
  published
    procedure AreaDeveSer6QuandoRetangulo_2X3;
    procedure AreaDeveSer9QuandoQuadrado_3X3;
    procedure AreaDeveSer20QuandoQuadradoRetangulo_4X5;
  end;

implementation

procedure TestTRetangulo.AreaDeveSer6QuandoRetangulo_2X3;
var
  retangulo: TRetangulo;
begin
  retangulo := TRetangulo.Create;
  retangulo.Altura := 2;
  retangulo.Largura := 3;
  CheckEquals(6, retangulo.Area);
end;

procedure TestTRetangulo.AreaDeveSer9QuandoQuadrado_3X3;
var
  retangulo: TQuadrado;
begin
  retangulo := TQuadrado.Create;
  retangulo.Altura := 3;
  CheckEquals(9, retangulo.Area);
end;

procedure TestTRetangulo.AreaDeveSer20QuandoQuadradoRetangulo_4X5;
var
  retangulo: TRetangulo;
begin
  retangulo := TQuadrado.Create;
  retangulo.Altura := 4;
  retangulo.Largura := 5;
  CheckEquals(20, retangulo.Area);
end;

initialization

RegisterTest(TestTRetangulo.Suite);

end.
